#! /bin/sh /usr/share/dpatch/dpatch-run
## 01_use_bin_sh.dpatch by Torsten Marek <shlomme@gmx.net>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Use /bin/sh to execute external programs instead of getting
## DP: the shell from $SHELL (See ticket #388)
#
# http://trac.lighttpd.net/trac/ticket/388

@DPATCH@
diff -urNad lighttpd-1.4.8~/src/proc_open.c lighttpd-1.4.8/src/proc_open.c
--- lighttpd-1.4.8~/src/proc_open.c	2005-08-11 00:26:39.000000000 +0200
+++ lighttpd-1.4.8/src/proc_open.c	2005-11-26 12:12:02.000000000 +0100
@@ -223,12 +223,13 @@
 /* {{{ proc_open */
 int proc_open(proc_handler_t *proc, const char *command) {
 	pid_t child;
-	const char *shell;
+	const char *shell = "/bin/sh";
 
-	if (NULL == (shell = getenv(SHELLENV))) {
+/*	if (NULL == (shell = getenv(SHELLENV))) {
 		fprintf(stderr, "env %s is required", SHELLENV);
 		return -1;
 	}
+*/
 
 	if (proc_open_pipes(proc) != 0) {
 		return -1;
